sources/settings-extensions/oci-defaults/src/de.rs

```diff
@@ -0,0 +1,126 @@
+use serde::de::Error;
+use serde::{Deserialize, Deserializer};
+
+/// This specifies that any non negative i64 integer, -1, and "unlimited"
+/// are the valid resource-limits. The hard-limit set to "unlimited" or -1
+/// and soft-limit set to "unlimited" or -1 are converted to u64::MAX in
+/// the spec file for the container runtime which ultimately represents
+/// unlimited for that resource
+pub(crate) fn deserialize_limit<'de, D>(deserializer: D) -> Result<i64, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    #[derive(Deserialize)]
```

Comment:
This is moved from an existing model, right? Do we need to also be removing code that is "moved" when we transform the model to use the extensions?

Reply:
Yes, I moved this from the existing model. I'll send another PR to remove the unused structs and code. For some reason, rustfmt didn't complain locally but failed during the build on Git. I need to debug that more, so to save time, I decided to handle it later.